[Membership] Membership CRUD API 추가

ybon1107 opened the pull request:
## PR Type
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Summary : 변경 내용
- **as-is**

- **to-be**
  - membership crud api 추가

## Issue Number
- close #83 
- close #94 

## Etc
추후 사용자 정보 받아와 권한 체크 로직 추가

ybon1107 committed:
feat: membership - membershipUser membership 연결#83

- membership_user 제약 조건 설정(season, user_id)
- membership_user membership join

ybon1107 committed:
feat: membership - membership create api#83

- membership create api 추가

ybon1107 committed:
feat: eureka - eureka flyway 업데이트#83

- membership_user alter

ybon1107 committed:
Merge pull request #10 from ybon1107/feat/membership/#83-membershipUser-entity-add

83 브랜치 origin dev 로 머지

ybon1107 committed:
feat: membership - membership get detail api#94

- read membership ADMIN

ybon1107 committed:
feat: membership - membership get detail current year api#94

- season Integer로 타입 변경
- read current season api 추가

ybon1107 committed:
feat: membership - membership search api#94

- minDiscount, maxDiscount 설정

ybon1107 committed:
feat: membership - membership update api#94

ybon1107 committed:
feat: membership - membership delete api#94

ybon1107 committed:
Merge pull request #11 from ybon1107/feat/membership/#94-membership-rud-api

94 브랜치 dev merge

ybon1107 committed:
Merge pull request #103 from ybon1107/dev

feat #83, #94 branch dev merge

Juneunbae commented on eureka-service/src/main/resources/db/migration/V29__create_partitioned_membership_user.sql:
```diff
@@ -0,0 +1,40 @@
+RENAME TABLE p_membership_user TO p_membership_user_legacy;
+
+-- 파티션 테이블 재생성
+CREATE TABLE p_membership_user (
```
인덴트 다시 한번 논의해봐야겠네요!

Juneunbae commented on membership-service/src/main/java/com/boeingmerryho/business/membershipservice/application/dto/query/MembershipSearchCondition.java:
```diff
@@ -0,0 +1,11 @@
+	Double maxDiscount,
+	Boolean isDeleted
+) {
+
```
다른 Dto 와 다르게 여기만 엔터있습니다

Juneunbae commented on membership-service/src/main/java/com/boeingmerryho/business/membershipservice/domain/entity/Membership.java:
```diff
@@ -30,19 +33,26 @@ public class Membership extends BaseEntity {
+			this.name = update.name();
+		if (update.discount() != null)
+			this.discount = update.discount();
+
```
여기도 엔터ㄱ ㅏ있습니다

Juneunbae commented on membership-service/src/main/java/com/boeingmerryho/business/membershipservice/domain/type/MembershipType.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Getter;
+
+@Getter
+public enum MembershipType {
```
ENUM 값은 알아볼 수 있어도 Description 을 붙여서 설명을 써주는게 좋습니다

ybon1107 committed:
Merge branch 'BoeingMerryHo:dev' into dev

Juneunbae reviewed (approved):
고생하셨습니다

ybon1107 commented:
실수로 resolve conversation 눌러서 은배님 댓글이 사라졌네요..!
flyway 관련해서 오류가 발생할 것을 방지해서 일단은 29 삭제하여 업데이트하겠습니다!
은배님 말씀주신 코드 컨벤션도 추가로 수정하겠습니다!

ybon1107 committed:
refactor: membership - refactoring membership#94

- code convention 에 맞춰 수정
- MembershipType enum description 추가
- V29 flyway sql 삭제

ybon1107 committed:
Merge pull request #12 from ybon1107/feat/membership/#94-membership-rud-api

refactor: membership - refactoring membership#94

ybon1107 committed:
Merge pull request #106 from ybon1107/dev

refactoring 추가

Juneunbae commented:
> 실수로 resolve conversation 눌러서 은배님 댓글이 사라졌네요..! flyway 관련해서 오류가 발생할 것을 방지해서 일단은 29 삭제하여 업데이트하겠습니다! 은배님 말씀주신 코드 컨벤션도 추가로 수정하겠습니다!

예본님 근데 멤버십 구매일 때도 선착순 개념이 들어가서 락 적용해야해요..!

haisley77 commented on membership-service/src/main/java/com/boeingmerryho/business/membershipservice/application/service/admin/MembershipAdminService.java:
```diff
@@ -0,0 +1,63 @@
+
+	@Transactional
+	public MembershipUpdateResponseServiceDto updateMembership(Long id, MembershipUpdateRequestServiceDto requestDto) {
+		validator.validateHasUpdatableFields(requestDto);
```
validator 책임 분리! 좋은 것 같아요

haisley77 commented on membership-service/src/main/java/com/boeingmerryho/business/membershipservice/infrastructure/repository/MembershipQueryRepositoryImpl.java:
```diff
@@ -0,0 +1,90 @@
+	public Page<Membership> search(MembershipSearchCondition condition, Pageable pageable) {
+		QMembership membership = QMembership.membership;
+
+		BooleanBuilder where = buildCondition(membership, condition);
```
메서드를 분리하니 훨씬 가독성이 좋아졌어요

haisley77 reviewed (approved):
코드가 클린해요 👍
